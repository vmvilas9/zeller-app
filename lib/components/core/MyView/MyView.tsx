import {View, ViewProps} from "react-native";
import React from "react";

export const MyView = ({style, ...props}: ViewProps) => {
    return <View style={[{backgroundColor: "#FFFFFF"}, style]} {...props} />
}
