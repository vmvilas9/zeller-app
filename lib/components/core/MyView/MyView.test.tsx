import React from 'react';
import { render } from '@testing-library/react-native';
import { MyView } from './MyView';

describe('MyView', () => {
    it('renders correctly', () => {
        const { toJSON } = render(<MyView />);
        expect(toJSON()).toMatchSnapshot();
    });

    it('applies default styles', () => {
        const { getByTestId } = render(<MyView testID="my-view" />);
        const view = getByTestId('my-view');
        expect(view.props.style).toContainEqual({ backgroundColor: '#FFFFFF' });
    });

    it('overrides styles with provided style prop', () => {
        const { getByTestId } = render(
            <MyView testID="my-view" style={{ padding: 10 }} />
        );
        const view = getByTestId('my-view');
        expect(view.props.style).toContainEqual({ padding: 10 });
    });

    it('merges provided styles with default styles', () => {
        const { getByTestId } = render(
            <MyView testID="my-view" style={{ padding: 10 }} />
        );
        const view = getByTestId('my-view');
        expect(view.props.style).toEqual(
            expect.arrayContaining([{ backgroundColor: '#FFFFFF' }, { padding: 10 }])
        );
    });

    it('passes down other props correctly', () => {
        const mockOnLayout = jest.fn();
        const { getByTestId } = render(
            <MyView testID="my-view" onLayout={mockOnLayout} />
        );
        const view = getByTestId('my-view');
        expect(view.props.onLayout).toBe(mockOnLayout);
    });
});
