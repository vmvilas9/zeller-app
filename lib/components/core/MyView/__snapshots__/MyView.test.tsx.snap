// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyView renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
      undefined,
    ]
  }
/>
`;
